dvc/config.py

```diff
@@ -598,25 +580,49 @@ def show(config, section, opt):
                 "option '{}.{}' doesn't exist".format(section, opt)
             )
 
-        logger.info(config[section][opt])
-
-    @staticmethod
-    def _merge(first, second):
-        res = {}
-        sections = list(first.keys()) + list(second.keys())
-        for section in sections:
-            first_copy = first.get(section, {}).copy()
-            second_copy = second.get(section, {}).copy()
-            first_copy.update(second_copy)
-            res[section] = first_copy
-        return res
+        return config[section][opt]
 
     @staticmethod
     def _lower(config):
-        new_config = {}
+        new_config = configobj.ConfigObj()
         for s_key, s_value in config.items():
             new_s = {}
             for key, value in s_value.items():
                 new_s[key.lower()] = str(value)
             new_config[s_key.lower()] = new_s
         return new_config
+
+    def get_configobj(self, level):
+        configs = {
+            self.LEVEL_LOCAL: self._local_config,
+            self.LEVEL_REPO: self._repo_config,
+            self.LEVEL_GLOBAL: self._global_config,
+            self.LEVEL_SYSTEM: self._system_config,
+        }
+
+        return configs.get(level, self._repo_config)
+
+    def add_fmt(self, fmt, name, url, level=None, force=False):
+        section = fmt.format(name)
+        self.set(
+            section, self.SECTION_REMOTE_URL, url, level=level, force=force
+        )
+
+    def remove_fmt(self, fmt, name, level=None):
+        section = fmt.format(name)
+        self.unset(section, level=level)
+
+    def modify_fmt(self, fmt, name, option, value, level=None):
+        section = fmt.format(name)
+        self.set(section, option, value, level=level)
+
+    def list_urls(self, regex, level=None):
+        ret = {}
+        config = self.get_configobj(level)
+        for section in config.keys():
+            r = re.match(regex, section)
+            if r:
+                name = r.group("name")
+                url = config[section].get(Config.SECTION_REMOTE_URL, "")
```

Comment:
rename SECTION_REMOTE_URL, otherwise it's a little bit misleading . It's not about remotes only anymore.